nslsii/__init__.py

```diff
@@ -568,6 +573,21 @@ def kafka_publisher_factory(name, start_doc):
             serializer=partial(msgpack.dumps, default=mpn.encode),
         )
 
+        def handle_publisher_exceptions(name_, doc_):
+            """
+            Do not let exceptions from the Kafka producer kill the RunEngine.
+            """
+            try:
+                kafka_publisher(name_, doc_)
+            except Exception as ex:
+                logger = logging.getLogger("nslsii")
+                logger.error(
```

Comment:
`logger.exception` takes in a message string and _automatically_ includes the current traceback below it to create an ERROR-level log record. Example:

```py
In [3]: try: 
   ...:     1/0 
   ...: except Exception: 
   ...:     logger.exception("Just minding my own business when all of a sudden") 
   ...:                                                                                                                                                                                       
Just minding my own business when all of a sudden
Traceback (most recent call last):
  File "<ipython-input-3-c10aadf263f7>", line 2, in <module>
    1/0
ZeroDivisionError: division by zero
```

Thus, I think these two logger calls should be combined: drop `logger.error(...)` and pass its message to `logger.exception`.

Reply:
combined these two and two more farther down